Modules/_decimal/_decimal.c

```diff
@@ -3411,6 +3364,55 @@ dict_get_item_string(PyObject *dict, const char *key, PyObject **valueobj, const
     return 0;
 }
 
+/*
+ * Fallback _pydecimal formatting for new format specifiers that mpdecimal does
+ * not yet support. As documented, libmpdec follows the PEP-3101 format language:
+ * https://www.bytereef.org/mpdecimal/doc/libmpdec/assign-convert.html#to-string
+ */
+static PyObject *
+pydec_format(PyObject *dec, PyObject *fmt, decimal_state *state)
+{
+    PyObject *result;
+    PyObject *pydec;
+    PyObject *u;
+
+    if (state->PyDecimal == NULL) {
+        PyObject *obj = PyImport_ImportModule("_pydecimal");
+        if (obj == NULL) {
+            return NULL;
+        }
+
+        state->PyDecimal = PyObject_GetAttrString(obj, "Decimal");
+        Py_DECREF(obj);
+
```

Comment:
Since this code is not limited to pure public C API, you can use `_PyImport_GetModuleAttrString()`.
```suggestion
        state->PyDecimal = _PyImport_GetModuleAttrString("_pydecimal", "Decimal");
```